src/components/profile/PreviewAs.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <aside class="preview-as">
+    <Button
+      v-if="viewAsFilter.active"
+      class="preview-as__button"
+      @click="deactivate"
+      ><Icon id="chevron-left" :width="17" :height="17"></Icon>Back</Button
+    >
+    <div v-if="viewAsFilter.active" class="preview-as__select-container">
+      <span>Previewing profile as</span>
+      <Select
+        class="privacy-select privacy-select--blue privacy-select--large"
+        :label="'Preview Profile As'"
+        :id="'preview-profile-selection'"
+        v-model="viewAsFilter.filter"
+        :options="viewAsOptions"
+        :expanededShowIcon="true"
+        :collapsedShowIcon="true"
+        @input="$emit('input', $event)"
+      />
+    </div>
+    <Button
+      v-else
+      @click="activate"
+      class="preview-as__button preview-as__activate-button"
+      ><Icon id="eye" :width="17" :height="17"></Icon><span>Preview As</span
+      ><Icon id="chevron-right" :width="17" :height="17"></Icon
+    ></Button>
+  </aside>
+</template>
+
+<script>
+import Icon from '@/components/ui/Icon.vue';
+import Button from '@/components/ui/Button.vue';
+import Select from '@/components/ui/Select.vue';
+
+export default {
+  name: 'PreviewAs',
+  components: { Select, Button, Icon },
+  props: {
+    viewAsFilter: Object,
+  },
+  methods: {
+    activate() {
+      this.viewAsFilter.active = true;
+    },
+
+    deactivate() {
+      this.viewAsFilter.active = false;
+      this.viewAsFilter.filter = null;
+    },
+  },
+  destroyed() {
+    this.deactivate();
+  },
+  data() {
+    return {
+      viewAsOptions: [
```

Comment:
pretty much the privacy settings. TODO would be rename/structure the icons. Also something for the SVG redo.